packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -37,12 +54,58 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 		};
 	}, [sharedCounter, setValue]);
 
+    function decrementCounter(): void {
+        sharedCounter.increment(-Number.parseInt(deltaValue, 10));
+        setDeltaValue("");
+	}
+
+    function incrementCounter(): void {
+		sharedCounter.increment(Number.parseInt(deltaValue, 10));
+        setDeltaValue("");
+	}
+
+    const inputSetValue = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        setDeltaValue(e.target.value);
+    }
+
 	return (
 		<Stack>
 			<StackItem>
 				<b>SharedCounter</b>
 			</StackItem>
-			<StackItem>Value: {value}</StackItem>
+			<StackItem>Value: {value} </StackItem>
+            <StackItem>
+                <TooltipHost
+					content="Decrememt counter by the delta-value."
+					id={decrementButtonTooltipId}
+				>
+                    <IconButton
+                            onClick={decrementCounter}
+                            disabled={deltaValue === ""}
+                            menuIconProps={{ iconName: "CalculatorSubtract" }}
+                            aria-describedby={decrementButtonTooltipId}
+                    />
+                </TooltipHost>
+
+                <TooltipHost
+                    content="Enter the delta value of your choice"
+                    id={inputFieldTooltipId}
+                >
+                    <input onChange={inputSetValue} placeholder="Enter Delta" value={deltaValue}/>
```

Comment:
Are we still getting any content enforcement here? We definitely still want to ensure users can only enter integer values into the field.